src/analyzer/protocol/http/HTTP.cc

```diff
@@ -1020,6 +1020,32 @@ void HTTP_Analyzer::Undelivered(uint64_t seq, int len, bool is_orig) {
     }
 }
 
+void HTTP_Analyzer::FlipRoles() {
+    analyzer::tcp::TCP_ApplicationAnalyzer::FlipRoles();
+
+    // If FlipRoles() is invoked after we've upgraded to something,
+    // don't do anything. This shouldn't happen as flipping of TCP
+    // connections currently happens before any data is transferred,
+    // but better safe than sorry.
+    if ( upgraded || pia )
+        return;
+
+    // If we haven't upgraded but saw requsts/replies, just bail for
+    // the rest of this connection. Again, this should never happen
+    // right now, but raise a weird in case it starts to happen.
+    if ( num_requests > 0 || num_replies > 0 ) {
+        Weird("HTTP_late_flip_roles");
+        SetSkip(true);
+        return;
+    }
+
+    // IsOrig() of the support analyzer has been updated, but we still
+    // need to change our state.
+    std::swap(content_line_orig, content_line_resp);
+    content_line_orig->SetSkipPartial(false);
+    content_line_resp->SetSkipPartial(true);
```

Comment:
These seems to assume which way we're flipping. That's in practice probably  valid, but in principle `FlipRoles()` could flip either way, so I think this should instead flip the partial flags, too, instead of hardcoding the new ones.